### PR TITLE
Add support for Nobrainer (RethinkDB)

### PR DESCRIPTION
**Pull request by zedtux**
* Add persistence for NoBrainer
* Add tests for NoBrainer
* Show the ORM version when running the tests
* Add a Dockerfile and a docker-compose.yml file allowing to run from anywhere very easily the test suite with `docker-compose run --rm aasm\'
* Moves the Mongoid database name definition in the gem loader (avoiding code repetition)
* Update the CHANGELOG.md and the README.md files
* Add Rails generator for NoBrainer

FYI I had to create new Appraisals for NoBrainer as [there is a conflict with Mongoid](https://github.com/nviennot/nobrainer/blob/master/lib/no_brainer/connection_manager.rb#L17).

This contribution is in the name of [Pharmony](https://github.com/Pharmony)!

### CODE REVIEW

**Comment by zedtux** [comment]
The builds for the Rubies are passing now! Ready for merge 😎

**Comment by anilmaurya** [comment]
@zedtux Thank you for your efforts 👍

I will review and merge in couple of days .

File: lib/aasm/persistence/no_brainer_persistence.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        #   foo = Foo.find(:first)
+        #   foo.aasm_state # => 1
+        #   foo.aasm_state = nil
+        #   foo.valid?
+        #   foo.aasm_state # => nil
+        #
+        def aasm_ensure_initial_state
+          AASM::StateMachineStore.fetch(self.class, true).machine_names.each do |state_machine_name|
+            attribute_name = self.class.aasm(state_machine_name).attribute_name.to_s
+            if self.class.fields.stringify_keys.keys.include?(attribute_name) && send(attribute_name).blank?
```
[the remark this answers is not in the document]
Comment (zedtux) [reply lib/aasm/persistence/no_brainer_persistence.rb]: I fully agree with you. I have to admit I hardly followed what was done with Mongoid which wasn't a good idea for everything. Thank you! 👍

File: spec/unit/persistence/no_brainer_persistence_spec.rb [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      # if you want to see the statements while running the spec enable the
+      # following line
+      # NoBrainer.configure do |config|
+      #   config.logger = Logger.new(STDERR)
+      # end
+    end
+
+    after do
+      NoBrainer.purge!
+    end
```
[the remark this answers is not in the document]
Comment (zedtux) [reply spec/unit/persistence/no_brainer_persistence_spec.rb]: Well then it's conflicting with your previous comment:

> Things that are not specific to NoBrainer should probably be in a separate pull request. [...]

I would have to update the `spec/unit/persistence/mongoid_persistence_spec.rb`, `spec/unit/persistence/mongoid_persistence_mutliple_spec.rb`, `spec/unit/persistence/no_brainer_persistence_spec.rb` and `spec/unit/persistence/no_brainer_persistence_mutliple_spec.rb` files.

File: spec/spec_helpers/redis.rb [hunk 3]
```diff
@@ -1,9 +1,12 @@
 # encoding: utf-8
+
 begin
   require 'redis-objects'
-  puts "redis-objects gem found, running Redis specs \e[32m#{'✔'}\e[0m"
```
[the remark this answers is not in the document]
Comment (zedtux) [reply spec/spec_helpers/redis.rb]: @anilmaurya do you want a spearated PR for those changes?

[the remark this answers is not in the document]
Comment (anilmaurya) [reply spec/spec_helpers/redis.rb]: @zedtux It make sense to send separate PR for changes which are not specific to NoBrainer.

Thank you @nviennot for helping in code review. I appreciate your review. 🏅

**Comment by zedtux** [comment]
Code updated and tests passed, thank you @nviennot

File: spec/spec_helpers/redis.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (zedtux) [reply spec/spec_helpers/redis.rb]: Okay, then I will split this part.

**Comment by anilmaurya** [comment]
@zedtux Could you resolve conflicts and update this PR ?

**Comment by anilmaurya** [comment]
@zedtux Thank you updating.
I will try to spare some time for debugging this issue.

**Comment by zedtux** [comment]
@anilmaurya if you need anything from me, please feel free to ask.

**Comment by anilmaurya** [comment]
@zedtux Can you help in fixing travis build for this PR 😊

Thank you for asking ♡

**Comment by anilmaurya** [comment]
Hi @zedtux ,

I am trying to run test suite through docker
It gives me following error:
```
docker-compose run --rm aasm
Pulling bundle (aasm/aasm:latest)...
ERROR: pull access denied for aasm/aasm, repository does not exist or may require 'docker login'
```

**Comment by zedtux** [comment]
You have to run first `docker-compose build aasm`.
Otherwise Docker tries to download the image from Docker hub, but I will let you manage the upload and the maintenance.

**Comment by zedtux** [comment]
Oh that’s great 👍🏻

**Comment by alto** [comment]
@zedtux Great OR. Cheers!

@anilmaurya Thanks for taking care of this!

**Pull request merged**